### PR TITLE
Insufficient Funds Case Handling

### PR DESCRIPTION
**Pull request by ayushguptacse14**
# Goals of PR

Core changes:

- In case of insufficient funds specular's validator and sequencer will log the error and continue instead of panicing and exiting

### CODE REVIEW

File: clients/geth/specular/rollup/services/sequencer/sequencer.go [hunk 1]
```diff
@@ -249,7 +249,7 @@ func (s *Sequencer) sequencingLoop(genesisRoot common.Hash) {
 			confirmedAssertion.CumulativeGasUsed,
 		)
 		if errors.Is(err, core.ErrInsufficientFunds) {
-			log.Crit("[Sequencer: sequencingLoop] Insufficient Funds to send Tx", "error", err)
+			log.Error("[Sequencer: sequencingLoop] Insufficient Funds to send Tx", "error", err)
 		}
 		if err != nil {
 			log.Error("[Sequencer: sequencingLoop] Can not create DA", "error", err)
```
Comment (LEAFERx) [inline clients/geth/specular/rollup/services/sequencer/sequencer.go:255]: redundant log here

Comment (ayushguptacse14) [reply clients/geth/specular/rollup/services/sequencer/sequencer.go]: Removed redundancy "[Sequencer: sequencingLoop]"

File: clients/geth/specular/rollup/services/validator/validator.go [hunk 2]
```diff
@@ -318,7 +321,8 @@ func (v *Validator) challengeLoop() {
 					ctx.lastValidatedAssertion.CumulativeGasUsed,
 				)
 				if errors.Is(err, core.ErrInsufficientFunds) {
-					log.Crit("[Validator: challengeLoop] Insufficient Funds to send Tx", "error", err)
+					log.Error("[Validator: challengeLoop] Insufficient Funds to send Tx", "error", err)
+					continue
```
[the remark this answers is not in the document]
Comment (ayushguptacse14) [reply clients/geth/specular/rollup/services/validator/validator.go]: Sure will handle challenges in seperate pr

File: clients/geth/specular/rollup/services/validator/validator.go [hunk 3]
```diff
@@ -92,7 +92,7 @@ func (v *Validator) tryValidateAssertion(lastValidatedAssertion, assertion *roll
 	// Validation succeeded, confirm assertion and advance stake
 	_, err := v.Rollup.AdvanceStake(assertion.ID)
 	if errors.Is(err, core.ErrInsufficientFunds) {
-		return fmt.Errorf("[Validator: tryValidateAssertion] Insufficient Funds to send Tx, err: %w", err)
```
[the remark this answers is not in the document]
Comment (ayushguptacse14) [reply clients/geth/specular/rollup/services/validator/validator.go]: Created variable errInsufficientFunds to handle insufficient funds error case

**Comment by ujvl** [comment]
handled in #107

**Pull request closed without merging**